### PR TITLE
mgr/dashboard: Add group field in nvmeof service form

### PR DESCRIPTION
**Pull request by afreen23**
Fixes https://tracker.ceph.com/issues/67223

[Screencast from 2024-08-20 14-54-41.webm](https://github.com/user-attachments/assets/6e38f993-ccc3-466a-af83-52c32c199d80)

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by cloudbehl — APPROVED** [review]
LGTM.

**Comment by nizamial09** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html [hunk 1]
```diff
@@ -105,6 +105,30 @@
...
+                     formControlName="group"
+                     (change)="onNvmeofGroupChange($event.target.value)">
+            </div>
+            <cd-help-text i18n>
+              The name of the gateway group.
+            </cd-help-text>
+            <span class="invalid-feedback"
+                  *ngIf="serviceForm.showError('service_id', frm, 'required')"
+                  i18n>This field is required.</span>
+          </div>
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html:129]: isn't group an optional field? if the group is not provided it should proceed to create with the pool name, wasn't that the behavior?

Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.html:129]: never mind, i got my answer through the cli
```
ceph orch apply nvmeof rbd
Invalid command: missing required parameter group(<string>)
```

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
Failing tests are unrelated.

**Pull request merged**